lib/sidekiq_prometheus/metrics.rb

```diff
@@ -130,26 +138,48 @@ class << self
   # @param types [Symbol] type of metric to register. Valid types: %w(counter gauge summary histogram)
   # @param name [Symbol] name of metric
   # @param docstring [String] help text for metric
-  # @param base_labels [Hash] Optional hash of base labels to use for every instance of this metric
+  # @param labels [Array] Optionally an array of labels to configure for every instance of this metric
+  # @param preset_labels [Hash] Optionally a Hash of labels to use for every instance of this metric
   # @param buckets [Hash] Optional hash of bucket values. Only used for histogram metrics.
-  def register(type:, name:, docstring:, base_labels: {}, buckets: nil)
+  def register(type:, name:, docstring:, labels: [], preset_labels: {}, buckets: nil)
     raise InvalidMetricType, type unless VALID_TYPES.include? type
 
-    custom_labels = SidekiqPrometheus.custom_labels[name]
+    # Aggregate all preset labels
+    all_preset_labels = {}
+    validate_type!(:arg_preset_labels, preset_labels, Hash) &&
+      all_preset_labels.merge!(preset_labels.dup)
+    validate_type!(:preset_labels, SidekiqPrometheus.preset_labels, Hash) &&
+      all_preset_labels.merge!(SidekiqPrometheus.preset_labels)
```

Comment:
I'm not sure the type checking here adds much. If neither is a Hash we'll still raise an error. I think this should just be:

```
preset_labels.merge! SidekiqPrometheus.base_labels if SidekiqPrometheus.base_labels
```